### DISCUSSION TITLE
Mssql connection timeout

### DISCUSSION THREAD

**Discussion by jayhyuk** [grafana/grafana#50915] [Data Sources]
As user we can we set timeout on mssql query foreach mssql datasource ?

**Comment by Trovalo** [comment]
The short answer is no, and I doubt it will ever be available as I think it will require low-level and specific customization just for one data source. 

As you may know, SQL Server has two types of timeouts (both to be managed by the client):
- Connection Timeout
This is a connection property (SqlConnection.ConnectionTimeout) → _The time (in seconds) to wait for a connection to open. The default value is 15 seconds._
- Command Timeout
This is a command property (SqlCommand.CommandTimeout) → _The time in seconds to wait for the command to execute._

Even if the `Connection Timeout` could be set in Grafana (as is a connection property like others already available) I doubt it will help as that's just the time to wait for a connection to open.
The `Command Timeout` must be set in the single command execution, so not even at panel level, at an even lower level.

**Comment by jayhyuk** [comment]
Hi,

Thanks for reponse

At first i exactly thing the same as you

But for the go library that grafana use


Connection timeout is effect to the command timeout , seem like it cut
connection if connection(connect+command) greater than this number




I think for this lib
there are dial timeout , connection timeout



Actually i already did , can share you real scenario if you want.






ในวันที่ ศ. 17 มิ.ย. 2022 18:00 น. Giovanni Luisotto <
***@***.***> เขียนว่า:

**Reply by Trovalo** [reply]
I'm not proficient in GO, but I may know the library you are talking about ([go-mssqldb](https://github.com/denisenkom/go-mssqldb)).
If that's the case, it may be a nice addition to the current config options available, the best way to find out is to make a PR that adds this parameter and see how it goes.

**Comment by jayhyuk** [comment]
Hi , I'm created PR ,
for testing this connection timeout which already attach test result
https://github.com/grafana/grafana/pull/51067

ในวันที่ ศ. 17 มิ.ย. 2022 เวลา 21:45 Giovanni Luisotto <
***@***.***> เขียนว่า:

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
